### PR TITLE
Reactivates Away Gate

### PR DESCRIPTION
**Pull request by SpadesNeil**
Thanks to @Datraen for helping to unfuck this.

### CODE REVIEW

File: maps/RandomZLevels/snowfield.dm [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+/obj/effect/floor_decal/derelict/d16
+	name = "derelict16"
+	icon_state = "derelict16"
+
+/obj/effect/spawner/snowfield // Hacky as all hell and causes slow round loads BUT IT FUCKING WORKS NOW, DOESN'T IT? -Spades
+	icon = 'icons/obj/flora/snowflora.dmi'
+	icon_state = "snowbush1"
+
+	New()
```
Comment (SpadesNeil) [inline maps/RandomZLevels/snowfield.dm]: This entire proc I've written is laggy as fuck and I need help unfucking it.

File: maps/RandomZLevels/snowfield.dm [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+				qdel(src)
+			if(11 to 30)
+				new /obj/structure/flora/bush(get_turf(src))
+			if(1 to 10)
+				new /obj/structure/flora/tree/pine(get_turf(src))
+			if(100)
+				var/junk = pick(/mob/living/simple_animal/hostile/samak, /mob/living/simple_animal/hostile/diyaab, /mob/living/simple_animal/hostile/shantak,
+							/mob/living/simple_animal/hostile/vore/bear/polar, /obj/effect/landmark/away)
+				new junk(get_turf(src))
+		qdel(src)
```
Comment (SpadesNeil) [inline maps/RandomZLevels/snowfield.dm]: This entire thing is laggy as fuck and I need help making it not lag like fuck.

**Comment by SpadesNeil** [comment]
Look even Travis agrees.

File: maps/RandomZLevels/snowfield.dm (hunk 2 above)
Comment (Screemonster) [inline maps/RandomZLevels/snowfield.dm]: Mentioned in discord, but iterating through every tile for a 1% chance of a mob, 10% of a tree and 20% of a bush seems like an inefficient way to go about it compared to, say...

place an animal on a random snow turf, X times
place a tree on a random snow turf, Y times
place a bush on a random snow turf, Z times

Could even cut overhead down further by having some of the trees and bushes fixed on the map 'cause it's not like anyone's gonna notice if it's _mostly_ random.

**Comment by SpadesNeil** [comment]
I need to make the snow map smaller. :\

**Comment by SpadesNeil** [comment]
Only problem remains is the random generation script doesn't work for the snow map and I cannot figure out why.

**Comment by ArgobargSoup** [comment]
I'm not familiar with Travis at all, but I looked at the Travis CI thingie, and took a look at what seemed ta be the error...

In /maps/RandomZLevels/snowfield.dm, lines 15 and 25 seem ta correspond to the Travis error.  Is the comment "//Pine is the most common" in lines 14 and 24 causing a bug?

**Comment by SpadesNeil** [comment]
That shouldn't actually change anything... :T

**Comment by ArgobargSoup** [comment]
So Travis says it's OK now... does it actually work now?

**Comment by SpadesNeil** [comment]
It works, but I am not satisfied with the final product due to lighting issues that I am having trouble fixing.

https://github.com/SpadesNeil/VOREstation/issues/623

Exact same issue as here. I cannot recall how I fixed it.

**Comment by ArgobargSoup** [comment]
Is there a secondary "refresh lighting" thingie that can be done without laggin' the server too much?

**Comment by Datraen** [comment]
As of my changes, ZAS and dynamic lighting overlays work on the away missions.

**Comment by SpadesNeil** [comment]
This is finally good to merge, I think.

**Pull request merged**